src/main/kotlin/org/elm/ide/intentions/AddQualifierIntention.kt

```diff
@@ -0,0 +1,126 @@
+package org.elm.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.psi.PsiElement
+import org.elm.lang.core.psi.*
+import org.elm.lang.core.psi.elements.*
+import org.elm.lang.core.psi.elements.Flavor.*
+import org.elm.lang.core.resolve.ElmReferenceElement
+import org.elm.lang.core.resolve.scope.ModuleScope
+import org.elm.lang.core.resolve.scope.VisibleNames
+import org.elm.lang.core.types.moduleName
+import org.elm.openapiext.isUnitTestMode
+import org.elm.openapiext.runWriteCommandAction
+import org.elm.openapiext.toPsiFile
+import org.jetbrains.annotations.TestOnly
+
+class AddQualifierIntention : ElmAtCaretIntentionActionBase<AddQualifierIntention.Context>() {
+
+    data class Context(
+            val candidates: List<String>,
+            val referenceName: String,
+            val qid: ElmQID
+    )
+
+    override fun getText() = "Qualify name"
+    override fun getFamilyName() = text
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        val file = editor.toPsiFile(project) as? ElmFile ?: error("no file: should not happen")
+        if (element.ancestors.any { it is ElmModuleDeclaration || it is ElmPortAnnotation || it is ElmImportClause }) {
+            return null
+        }
+
+        val qid = element.parentOfType<ElmQID>() ?: return null
+        if (qid.isQualified) return null
+
+        val typeRef = qid.parentOfType<ElmTypeRef>()
+        if (typeRef != null) {
+            return makeContext(file, qid, typeRef, ModuleScope.getReferencableTypes(file))
+        }
+
```

Comment:
I noticed that the intention is not available for `BadStatus` in the following:

```elm
import Http

decodeErrors : Http.Error -> List String
decodeErrors error =
    case error of
        BadStatus response ->
            [ "bad" ]

        err ->
            [ "Server error" ]
```

The problem is that the `qid`'s parent in this case is `ElmUnionPattern` not `ElmValueExpr`.